platforms/core-configuration/kotlin-dsl-provider-plugins/src/main/kotlin/org/gradle/kotlin/dsl/provider/plugins/precompiled/tasks/GeneratePrecompiledScriptPluginAccessors.kt

```diff
@@ -409,12 +429,53 @@ abstract class GeneratePrecompiledScriptPluginAccessors @Inject internal constru
     }
 
     private
-    fun buildLogicClassPath(): ClassPath =
-        services.get<CachedClasspathTransformer>().transform(
-            DefaultClassPath.of(runtimeClassPathFiles),
-            CachedClasspathTransformer.StandardTransform.BuildLogic
+    fun buildLogicClassPath(gradle: Gradle): ClassPath {
+        // Ideally we would pass already instrumented classpath to a task and then just export it to the classloader.
+        // But since we do some artifact transform caching via BuildService, that would add some complexity when wiring GeneratePrecompiledScriptPluginAccessors task.
+        val dependencyManagementServices = gradle.serviceOf<DependencyManagementServices>()
+        val fileCollectionFactory = gradle.serviceOf<FileCollectionFactory>()
+        val dependencyResolutionServices = dependencyManagementServices.create(
+            gradle.serviceOf<FileResolver>(),
+            fileCollectionFactory,
+            gradle.serviceOf<DependencyMetaDataProvider>(),
+            UnknownProjectFinder("Project dependencies are not allowed at GeneratePrecompiledScriptPluginAccessors resolution"),
```

Comment:
❔ I'm not familiar with these. Does this work with project dependencies of the precompiled script plugins? E.g. I can have a plugin project that depends on some other project providing common utilities, etc.

Reply:
At this point we already resolved original classpath so we don't original dependencies anymore. So we actually instrument just _file dependncies_, this can be seen here https://github.com/gradle/gradle/blob/f5b71d24c100b2536f29a14744d62a24b623d90d/platforms/core-configuration/kotlin-dsl-provider-plugins/src/main/kotlin/org/gradle/kotlin/dsl/provider/plugins/precompiled/tasks/GeneratePrecompiledScriptPluginAccessors.kt#L463-L473, where we just create a file dependencies.

We would need to get an instrumententated view of a configuration we pass to the `GeneratePrecompiledScriptPluginAccessors`. Butsince we do some artifact transform caching via BuildService, that would add some complexity when wiring GeneratePrecompiledScriptPluginAccessors task.